client/src/Components/Nav.js

```diff
@@ -57,12 +64,12 @@ const Nav = () => {
                     variant="outlined"
                     />
                 <ul className={classes.ul}>
-                    <NavLink className={classes.a} to="/profile" activeClassName="active"><li className={classes.li}>Share music</li></NavLink>
-                    <NavLink className={classes.a} to="/discover" activeClassName="active"><li className={classes.li}>Discover</li></NavLink>
-                    <NavLink className={classes.a} to="/messages" activeClassName="active"><li className={classes.li}>Messages</li></NavLink>
+                    <a href="./" className={classes.a}><li className={classes.li}><NavDialog label={"Share Music"} /></li></a>
+                    <a href="./" className={classes.a}><li className={classes.li}><NavDialog label={"Discover"} /></li></a>
+                    <a href="./" className={classes.a}><li className={classes.li}><NavDialog label={"Massages"} /></li></a>
```

Comment:
I think you still want Discover and Messages to be routes. And for the ShareMusic dialog, you want it to be a button, not an anchor tag. Maybe change the placement to make it more clear that Share Music is different then the other items on the nav

Reply:
Also, does the dialog actually open? Usually, the logic for closing/opening is in the parent component and `open` is passed as a prop

Reply:
yaa. please check now.